superset/db_engine_specs/mssql.py

```diff
@@ -50,8 +50,10 @@ def epoch_to_dttm(cls):
         return "dateadd(S, {col}, '1970-01-01')"
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
-        return "CONVERT(DATETIME, '{}', 126)".format(dttm.isoformat())
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
+        if target_type.upper() in ("DATE", "DATETIME", "SMALLDATETIME", "TIMESTAMP"):
+            return f"CONVERT(DATETIME, '{dttm.isoformat()}', 126)"
```

Comment:
I think `DATE` and `DATETIME` should be separated here:
```python
        if target_type.upper() in ("DATETIME", "SMALLDATETIME", "TIMESTAMP"):
            return f"CONVERT(DATETIME, '{dttm.isoformat()}', 126)"
        elif target_type.upper()  == "DATE"):
            return f"CONVERT(DATE, '{dttm.isoformat()}', 23)" 
```

Reply:
@villebro I addressed your comment.